engine/src/main/java/org/hibernate/validator/internal/engine/ConstraintViolationImpl.java

```diff
@@ -329,10 +326,10 @@ public String toString() {
 	private int createHashCode() {
 		int result = interpolatedMessage != null ? interpolatedMessage.hashCode() : 0;
 		result = 31 * result + ( propertyPath != null ? propertyPath.hashCode() : 0 );
+		result = 31 * result + ( constraintDescriptor != null ? constraintDescriptor.hashCode() : 0 );
 		result = 31 * result + System.identityHashCode( rootBean );
 		result = 31 * result + System.identityHashCode( leafBeanInstance );
 		result = 31 * result + System.identityHashCode( value );
-		result = 31 * result + ( constraintDescriptor != null ? constraintDescriptor.hashCode() : 0 );
 		result = 31 * result + ( messageTemplate != null ? messageTemplate.hashCode() : 0 );
 		result = 31 * result + ( rootBeanClass != null ? rootBeanClass.hashCode() : 0 );
```

Comment:
Shouldn't we remove it from the hashCode too?

Reply:
removed this one as well.